.github/actions/create-pr/action.yml

```diff
@@ -0,0 +1,92 @@
+name: "Create PR custom action"
+description: "Create a PR and a temporary branch, close duplicates"
+
+# PROCESS
+#
+# 1. Setup git client using Powertools for AWS Lambda bot username
+# 2. Pushes staged files to a temporary branch
+# 3. Creates a PR from temporary branch against a target branch (typically trunk: develop, main, etc.)
+# 4. Searches for duplicate PRs with the same title
+# 5. If duplicates are found, link to the most recent one, close and delete their branches so we keep a single PR
+# 6. In the event of failure, we delete the now orphaned branch (if any), and propagate the failure
+
+# REQUIREMENTS
+# You need to enable "Allow GitHub Actions to create and approve pull requests" in your repository settings
+# You need to have "skip-changelog" label in your repository to skip changelog generation.
+
+# USAGE
+#
+# - name: Create PR
+#   id: create-pr
+#   uses: ./.github/actions/create-pr
+#   with:
+#     files: "CHANGELOG.md"
+#     temp_branch_prefix: "ci-changelog"
+#     pull_request_title: "chore(ci): changelog rebuild"
+#     github_token: ${{ secrets.GITHUB_TOKEN }}
+# - name: Step to demonstrate how to access outputs (no need for this)
+#   run: |
+#     echo "PR number: ${PR_ID}"
+#     echo "Branch: ${BRANCH}"
+#   env:
+#     PR_ID: ${{ steps.create-pr.outputs.pull_request_id}}
+#     BRANCH: ${{ steps.create-pr.outputs.temp_branch}}
+
+inputs:
+  files:
+    description: "Files to add separated by space"
+    required: true
+  temp_branch_prefix:
```

Comment:
Couldn't this be generated?

Reply:
Strictly speaking, yes, however having a human readable prefix helps maintainers. Refer to [this comment](https://github.com/aws-powertools/powertools-lambda-typescript/pull/2006/files/3c849206fb967799e5141323e8f1521892e4a678#r1478038479) as to why these are inputs defined by the caller workflow.

Reply:
we could hard code the prefix for now, since only pt and ts teams use it and they can be the same.

Reply:
I think it's still useful to have a prefix defined by the caller:
i.e. `ci-version-???` or `ci-bump-???` for branches created as part of version bump vs `ci-docs-???` for branches created as part of the docs layer version (this latter will come in a future PR).

If for some reason the workflow(s) fail and leave dangling branches it'll be easier for us to tell them apart instead of having to scan the list, open each one, and see what's in it (which we would have to do if the name is opaque).

Reply:
Keeping this as an input option for now.